cli/dts/lib.deno.unstable.d.ts

```diff
@@ -129,6 +129,70 @@ declare namespace Deno {
     speed: number | undefined;
   }
 
+  /** All possible types for interfacing with foreign functions */
+  export type NativeType =
+    | "void"
+    | "u8"
+    | "i8"
+    | "u16"
+    | "i16"
+    | "u32"
+    | "i32"
+    | "u64"
+    | "i64"
+    | "usize"
+    | "isize"
+    | "f32"
+    | "f64";
+
+  type NativeTypeToJsType<T> = T extends "void" ? void
+    : T extends
+      | "u64"
+      | "i64"
+      | "usize"
+      | "isize" ? bigint
+    : T extends
+      | "u8"
+      | "i8"
+      | "u16"
+      | "i16"
+      | "u32"
+      | "i32"
+      | "f32"
+      | "f64" ? number
+    : never;
+
+  type MapParametersToJsType<P extends NativeType[]> = {
+    [I in keyof P]: NativeTypeToJsType<P[I]>;
+  };
```

Comment:
These automatic mapping of C types to typescript types is untested as far as I can see.

I wonder if we could do a minimal first pass of this that did not include this mapping, but rather a minimal interface for opening DLLs and calling functions in them. Callers would have to be careful, but if they're calling `dlopen` they probably know what they're doing.

Reply:
The mapping works to a certain extent but due to limitations with typescript the spread operator doesn't work on tuple types which makes this type mapping not work properly (apart from the return type). We could totally do without it for the first iteration, and the people using ffi will most likely be wrapping all native calls with custom types and functions anyways so that should be fine.

Reply:
Would you mind removing / simplifying it?

Reply:
sure 👍